webportal/src/app/job/job-submit/job-submit.component.js

```diff
@@ -30,6 +30,15 @@ const jobSubmitHtml = jobSubmitComponent({
   loading: loadingComponent
 });
 
+const isValidJson = (str) => {
+  try {
+    JSON.parse(str);
+    return true;
+  } catch (e) {
+    return false;
+  }
+};
```

Comment:
shall we implement it in rest server?

Reply:
no. If the uploaded file isn't a valid json, ajax can not send it as a json object to rest api.

Reply:
agree we should check the json format here. but it looks like we need to do it on rest-server as well, since in the future users can submit jobs through restAPIs

---
In reply to: [159381738](https://github.com/Microsoft/pai/pull/162#discussion_r159381738) [](ancestors = 159381738)

Reply:
Agree. One thing is not print the full stack, but just show the exception type and message.

Reply:
@fanyangCS rest-server's several middlewares have checked json format:
1. if user specified `Content-Type: application/json` header, [`Joi`](https://github.com/Microsoft/pai/blob/master/rest-server/src/config/job.js#L22) will make sure it's a valid json.
2. if `Content-Type: application/json` is not set in header, [`bodyParse`](https://github.com/Microsoft/pai/blob/master/rest-server/src/config/express.js#L36) will try to parse the request body to json and return error message if it's not a valid json.

what #161 met is error occurs in browser's `JSON.parse()` so the rest code wasn't executed.